hilo.rb

```diff
@@ -1,30 +1,42 @@
-# predefined number of guesses allowed
-guesses = 6
-## computer secretly picks number between 1 and 100 inclusive
-number = rand(1...101)
-
-# conditional logic that is run as long as user has remaining guesses left
-while guesses > 0
-    # get user input
-    puts "Guess the number between 1 & 100 (inclusive):"
-    # store user input
-    guess = gets.chomp.to_i
-    # if user guesses correctly, game is won (guess = number)
-    if (guess == number)
-        puts "You win! #{number} was the number!"
+# frozen_string_literal: true
+
+# Ruby HiLo console game where the user tries to guess a random number between 1 and 100.
+class HiLoGame
+  def initialize
+    @guesses = 6
+    @number = rand(1..100)
+  end
+
+  def play
+    while @guesses.positive?
+      puts 'Guess the number between 1 & 100 (inclusive):'
+      guess = gets.chomp.to_i
+
+      if correct_guess?(guess)
+        puts "You win! #{@number} was the number!"
         break
-    # if guess too high, decrement number of guesses left and tell them their guess was too high
-    elsif guess > number
-        guesses -= 1
-        puts "Your guess is too high!"
-    # if guess too low, decrement number of guesses left and tell them their guess was too low
-    elsif guess < number
-        guesses -=1
-        puts "Your guess is too Low!"
+      elsif too_high?(guess)
+        puts 'Your guess is too high!'
+      elsif too_low?(guess)
+        puts 'Your guess is too Low!'
+      end
+
+      @guesses -= 1
     end
-end
+    puts "You ran out of guesses! The number was #{@number}" if @guesses.zero?
+  end
+
+  def too_high?(guess)
```

Comment:
These methods should be private so that they can't be accessed outside this class. It helps when working with larger codebases to know which methods are designed to be used by other things and which are strictly internal.